src/Atmos/Model/filters.jl

```diff
@@ -51,3 +46,78 @@ function compute_filter_result!(
         filter_state.moisture.ρq_ice += aux.ref_state.ρq_ice
     end
 end
+
+struct AtmosSpecificFilterPerturbations{M} <: AbstractFilterTarget
+    atmos::M
+end
+
+vars_state_filtered(target::AtmosSpecificFilterPerturbations, FT) =
+    vars_state_filtered(target.atmos, FT)
+
+function compute_filter_argument!(
+    target::AtmosSpecificFilterPerturbations,
+    filter_state::Vars,
+    state::Vars,
+    aux::Vars,
+)
+
+    ρ_inv = 1 / state.ρ
+    filter_state.ρ = state.ρ - aux.ref_state.ρ
```

Comment:
I don't think we should filter density in the specific filter.

Reply:
I'm not sure how we can get `state.ρ` in `compute_filter_result!` if it's not filtered. Might need to rope in @mwarusz.

Reply:
In `compute_filter_result!` the `state` output argument comes in filled with the all state variables so it should contain `state.ρ`.